nlp_basics_articles/Paper_P_Lewis_2021.md

```diff
@@ -0,0 +1,198 @@
+
+## Highlights of this research
+- Proposed RAG models that use both parametric memory and non-parametric memory (RAG-Sequence model, RAG-Token model)
+    - Parametric: Feed **specific dataset** for the model to learn
+    - Non-parametric: Feed **general dataset** for the model to learn (Wikipedia dump in this paper)
+- Experimented on several benchmarks
+- Proposed models perform better than the SoTA methods
+
+## RAG (Retrieval-Augmented Generation)
+The RAG model used in this paper is illustrated in Figure 1.
+<p align="center">
+    <img width="600" alt="Screenshot 2024-12-02 at 01 06 08" src="https://github.com/user-attachments/assets/7fef3192-51dc-44b2-a2ed-11d0e2920aec">
+</p>
+<p align="center">
+Figure 1. RAG
+</p>
+
+Components in the model architecture:
+- Input (Query): Input to the model. (Question in QA task.)
+- Retriever: Retrieves relevant information from the database based on the input.
+- Database: Stores all given documents.
+- Documents: Subset of documents. $k$ documents are retrieved to be fed to the generator. (Also known as chunks in other works)
+- Previous tokens: Outputs from the model that was produced before this iteration. (Incomplete sentence)
+- Generator: Generates a current token based on given Documents, Input, and Previous tokens.
+- Current Token: Output from the generator. (A token following the incomplete sentence)
+
+## Proposed models and their architecture
+The authors propose two models: RAG-Sequence model and RAG-Token model. 
+They both have basically the same structure. The only difference is in how they make current token candidates and how they marginalize them. 
+**RAG-Sequence model** use the same document to predict each target token (current token).
+**RAG-Token model** predict each target token based on a different document.
+
+The difference between the two models is illustrated in Figure 2.
```

Comment:
I'm confused... which model are we using for our project right now? :joy:

Reply:
Hmm.. I actually don't know... I will think about it!